src/lib/format/l20n/ast/parser.js

```diff
@@ -101,16 +104,16 @@ class ParseContext {
     return entity;
   }
 
-  getValue(ch = this._source[this._index], optional = false) {
+  getValue(ch = this._source[this._index], index = false, required = true) {
```

Comment:
I can't fully reuse it because there's a semantic difference between the value has an index and the value is required, but I can reuse it where `hasIndex` makes it required.